plugins/woocommerce/src/LayoutTemplates/LayoutTemplateRegistry.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace Automattic\WooCommerce\LayoutTemplates;
+
+use Automattic\WooCommerce\Admin\BlockTemplates\BlockTemplateInterface;
+
+/**
+ * Layout template registry.
```

Comment:
We made a decision a while back to try distance ourselves from the "block" terminology for things that aren't inherently tied to blocks. Our layout templates really, at the heart of it, aren't dependent on blocks. We could interpret the layout template and instantiate non-block components.

`BlockTemplateRegistry` isn't used outside of the new product editor. I will deprecate it in this PR to make it clear that `LayoutTemplateRegistry` is the replacement.

I personally don't think that in the context of what we are implementing, we have any distinction between "block template" and "layout template". It is a naming change to make the intent clearer and hopefully prevent confusion between the use of blocks in the product editor (and other similar usage in the future for other screens such as orders management) and the idea of block templates used by the WP core editor implementation.

I also used this as an opportunity to rename things since the contact for registering template was changing (from passing in an instance to passing in a class name).

Reply:
> BlockTemplateRegistry isn't used outside of the new product editor. I will deprecate it in this PR to make it clear that LayoutTemplateRegistry is the replacement.

Actually, since we aren't using the `LayoutTemplateRegistry` in the new product editor yet in this PR (that will be handled in #40620), it makes more sense to do the deprecation in that follow-up.

Reply:
>We made a decision a while back to try distance ourselves from the "block" terminology for things that aren't inherently tied to blocks. Our layout templates really, at the heart of it, aren't dependent on blocks. We could interpret the layout template and instantiate non-block components.

Thanks for that explanation!  That makes sense to me.  Does this mean we'll also be renaming methods like `add_block` and other block references so that layouts are not tied to any block terminology?

Reply:
Yes, that is the plan... we have a chore in the backlog to take care of this. See #42885